### PR TITLE
Cypress tests for commenting feature

### PR DESCRIPTION
**Pull request by rishabhsaxena**
## How Has This Been Tested?
-  Test if comment pins are rendered for both modes (edit/published) and multiple users

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: collab-cypress-tests 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 53.39 **(0)** | 34.98 **(-0.01)** | 31.51 **(0)** | 53.94 **(0)**
 :red_circle: | app/client/src/pages/Editor/ToggleModeButton.tsx | 68.31 **(-0.48)** | 25.37 **(-0.78)** | 43.33 **(0)** | 66.92 **(-0.52)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 37.72 **(-0.88)** | 36.21 **(0)** | 55.08 **(-0.27)**</details>

### CODE REVIEW

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+const commentsLocators = require("../../../../locators/commentsLocators.json");
```
Comment (pranavkanade) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js:1]: Should we consider adding test to check the `new comment` indicator? I've seen people complain about it several times.

Comment (rishabhsaxena) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js:1]: Actually that was due to a pending functionality. Though this test is useful, also doesn't add much overheads to our test run times.

**Review by pranavkanade — DISMISSED** [review]

**Review by pranavkanade — APPROVED** [review]
LGTM! A suggestion - It'll be great if we could break this case down into smaller ones. That'll make it easy to debug, if anything break in future.

**Review by pranavkanade — APPROVED** [review]
LGTM!

**Comment by rishabhsaxena** [comment]
@pranavkanade
Added a task, would pick this up once the fix is merged:
https://github.com/appsmithorg/appsmith/issues/6270

**Review by pranavkanade — DISMISSED** [review]

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js [hunk 2]
```diff
@@ -55,6 +55,8 @@ describe("Comments", function() {
 
   it("new comments can be created after switching to comment mode", () => {
     return cy.wrap(null).then(async () => {
+      // wait for the page to load
+      cy.get(commonLocators.canvas);
```
Comment (pranavkanade) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js:59]: QQ: does this ensure that this test case will get pass at this point every time? Or shall we just add a hard wait?

**Review by pranavkanade — APPROVED** [review]
single comment, rest looks good

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js (hunk 2 above)
Comment (rishabhsaxena) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js:59]: So `cy.get` makes cypress retry by default until a timeout or the element if found, so I guess that's preferable over a specific wait time for the UI

**Review by pranavkanade — APPROVED** [review]
approving for tests

**Pull request merged**